java/src/jmri/jmrix/AbstractUsbConnectionConfig.java

```diff
@@ -0,0 +1,553 @@
+                    public void focusGained(FocusEvent e) {
+                    }
+                });
+                connectionNameField.addActionListener(new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        if (!adapter.getSystemConnectionMemo().setUserName(connectionNameField.getText())) {
+                            JOptionPane.showMessageDialog(null, Bundle.getMessage("ConnectionNameDialog", connectionNameField.getText()));
+                            connectionNameField.setText(adapter.getSystemConnectionMemo().getUserName());
+                        }
+                    }
+                });
+                connectionNameField.addFocusListener(new FocusListener() {
+                    @Override
+                    public void focusLost(FocusEvent e) {
+                        if (!adapter.getSystemConnectionMemo().setUserName(connectionNameField.getText())) {
+                            JOptionPane.showMessageDialog(null, Bundle.getMessage("ConnectionNameDialog", connectionNameField.getText()));
+                            connectionNameField.setText(adapter.getSystemConnectionMemo().getUserName());
+                        }
+                    }
+
+                    @Override
+                    public void focusGained(FocusEvent e) {
+                    }
+                });
+            }
+            portBox.addFocusListener(new FocusListener() {
+                @Override
+                public void focusGained(FocusEvent e) {
+                    refreshPortBox();
+                }
+
+                @Override
+                public void focusLost(FocusEvent e) {
+                }
+
+            });
+
+            for (String i : options.keySet()) {
+                final String item = i;
+                if (options.get(i).getComponent() instanceof JComboBox) {
+                    ((JComboBox<?>) options.get(i).getComponent()).addActionListener((ActionEvent e) -> {
+                        adapter.setOptionState(item, options.get(item).getItem());
+                    });
+                }
+            }
+            init = true;
+        }   // if (!init)
+    }   // checkInitDone()
+
+    @Override
+    public void updateAdapter() {
+        log.debug("*	updateAdapter()");
+    }
+
+    protected UserPreferencesManager p = InstanceManager.getDefault(UserPreferencesManager.class);
+    protected JComboBox<String> portBox = new JComboBox<>();
+    protected JLabel portBoxLabel;
+
+    @Override
+    public String getInfo() {
+        log.debug("*	getInfo()");
+        String t = (String) portBox.getSelectedItem();
+        if (t != null) {
+            return t;
+        } else if ((adapter != null) && (adapter.getCurrentPortName() != null)) {
+            return adapter.getCurrentPortName();
+        }
+
+        return JmrixConfigPane.NONE;
+    }
+
+    @SuppressWarnings("UseOfObsoleteCollectionType")
```

Comment:
Unless an API requires the use of Vectors, you don't need to use them.

Reply:
Gladly! Total pain copying Lists to Vectors. The code I plagiarized^h^h^h^h^h copied (<java/src/jmri/jmrix//AbstractSerialConnectionConfig.java>) used vectors… glad I don't have to. Thx!